.github/workflows/ci-core.yml

```diff
@@ -53,9 +54,28 @@ jobs:
         id: ignore-filter
         run: echo "changes=true" >> $GITHUB_OUTPUT
 
+      - name: Filter other changes
+        uses: ./.github/actions/filter-changes
+        id: filter
+        with:
+          # Filter files that match **every** pattern in this list
+          every-filter: |
+            - '**'                     # Initially include all files
+            - '!**/.github/**'         # Exclude all changes under the .github directory
+            - '!integration-tests/**'  # Exclude all changes under the integration-tests directory
+          # Filter files that match **any** of these patterns
+          any-filter: |
+            - '.github/workflows/ci-core.yml'
+            - '.github/actions/golangci-lint/**'
+            - '.github/actions/filter-changes/**'
+            - '.github/actions/setup-go/**'
+            - '.github/actions/setup-nodejs/**'
+            - '.github/actions/setup-postgres/**'
+            - '.github/actions/setup-solana/**'
+            - '.github/actions/setup-wasmd/**'
+
   golangci:
-    # We don't directly merge dependabot PRs, so let's not waste the resources
-    if: ${{ (github.event_name == 'pull_request' ||  github.event_name == 'schedule') && github.actor != 'dependabot[bot]' }}
+    if: github.actor != 'dependabot[bot]' && ((github.event_name != 'pull_request' && github.event_name != 'merge_group') || needs.filter.outputs.not_ignored_changes == 'true')
```

Comment:
Why is it OK to drop the explicit `schedule` logic here? Are we assuming the `event_name` based on not matching `pull_request` or `merge_group`?

Reply:
@jmank88 this will match every event that the workflow supports except `pull_request` and `merge_group`. So it will match `schedule`, `push` and `workflow_dispatch`. This means it will always run on these events (unless started by dependabot)

Reply:
But why change this? IIUC we did not trigger on `push` or `workflow_dispatch` before, and we did not intend to trigger on any events that github may add in the future.

Reply:
@jmank88 this workflow allows to trigger `push` and `workflow_dispatch`. I believe we should always run `golangci` for these events, similar to how we handle unit tests (see `core`). Executing `golangci` in these instances would be consistent.

Reply:
I still don't understand why, and I am not confident it will even work as expected. The linter is run in two different modes depending on context, and we have configured it very carefully. We have a nightly run that invokes a full linter analysis, and we have PRs that look only for new issues based on the diff. I can see a case for `workflow_dispatch`, but `push` just seems like wasteful noise.
cc @HenryNguyen5

Reply:
@jmank88 ok. I reverted it and just added the condition for PR event. `golangci` will run in the same way as before on `schedule` event.